docs/src/docs/user-guide.adoc

```diff
@@ -407,7 +459,10 @@ Jsonb jsonb = JsonbBuilder.create(nillableConfig);
 
 ==== Custom instantiation
 
-By default, public default no-arguments constructor is required for deserialization. In many scenarios this requirement is too strict. JSON-B provides @JsonbCreator annotation which can be used to annotate a custom constructor with parameters or a static factory method used to create a class instance.
+By default, public default no-arguments constructor is required for deserialization of the classes.
+In the case of records, it is required to have only default constructor, which was generated for the record.
```

Comment:
This seems to rule out having a compact constructor to override the default constructor that is generated for a record. Was that the intent? I can see where you might want to avoid the possibility that the compact constructor reassigns a record component. But I think it's more likely users would include the compact constructor for purposes like validating the supplied values.

Here is one idea if we want to conditionally allow the compact constructor for record:

```suggestion
By default, a public default no-arguments constructor is required for deserialization of the class.
In the case of records, the record's default constructor is required for deserialization of the record. The record can include a compact constructor if the compact constructor does not reassign any of the record components. 
```